src/plugins/intel_cpu/src/transformations/transformation_pipeline.cpp

```diff
@@ -852,6 +855,15 @@ void Transformations::PostLpt() {
     }
     CPU_REGISTER_PASS_COMMON(postLPTPassManager, ov::pass::transpose_sinking::TSShapeOfForward);
     CPU_REGISTER_PASS_COMMON(postLPTPassManager, StatefulSDPAFusion);
+    CPU_REGISTER_PASS_X64(postLPTPassManager, ov::pass::RMSFusion, false);
+    CPU_REGISTER_PASS_X64(postLPTPassManager, ov::intel_cpu::DecomposeRMSNorm);
+    CPU_SET_CALLBACK_X64(postLPTPassManager,
+        [](const std::shared_ptr<const ov::Node>& node) -> bool {
+            std::string errorMsg;
+            return node::RMSNorm::isSupportedOperation(node, errorMsg);
+        },
+        ov::intel_cpu::DecomposeRMSNorm);
```

Comment:
the logic here is complex & error prone due to that `RMSFusion` is calling tranformation_callback on raw pattern instead of new RMS node. 
Actually we can add an input parameter to constructor of `RMSFusion` to control where it calls tranformation_callback. so that we can avoid `DecomposeRMSNorm`

Reply:
GPU uses the callback and assume the type of the callback's parameter is `convert`. So the callback semantics should not be changed/moved. BTW, if RMSNorm could be generated in the frontend, in CPU plugin there still needs decomposition, such as not supported isa.